### PR TITLE
feat: render prebinding data in preview mode [SPA-2923]

### PR DESCRIPTION
**Pull request by Spring3**

### CODE REVIEW

File: packages/experience-builder-sdk/src/core/preview/PrebindingManager.ts [hunk 1]
```diff
@@ -0,0 +1,547 @@
+import {
```
Comment (SofiaMargariti) [inline packages/experience-builder-sdk/src/core/preview/PrebindingManager.ts:1]: Would it make sense to keep a minimal version of the prebindingManager that only includes:
* `storePrebindingDefinitions`
* `storeParameterDefinitions`
* `getHoistedParameterByNodeId`

Given that the one in the Web App has already diverged from this one, it just makes things more complicated to keep them in sync.

Comment (Spring3) [reply packages/experience-builder-sdk/src/core/preview/PrebindingManager.ts:1]: yes, it makes sense 👍

Comment (Spring3) [reply packages/experience-builder-sdk/src/core/preview/PrebindingManager.ts:1]: I will align it with the latest version from the web app I think

File: packages/experience-builder-sdk/src/core/preview/assemblyUtils.spec.ts [hunk 2]
```diff
@@ -84,44 +80,104 @@ describe('resolvePattern', () => {
...
-      );
+      expect(result.children[0].children[0]).toEqual({
+        ...assemblyEntry.fields.componentTree.children[0].children[0],
+        displayName: undefined,
+        parameters: undefined,
+        pattern: {
+          nodeIdOnPattern: 'test-custom-component-id',
+          parentPatternNodeId: undefined,
+          prefixedNodeId: 'test-custom-component-id',
+        },
```
Comment (Spring3) [inline packages/experience-builder-sdk/src/core/preview/assemblyUtils.spec.ts:99]: wanted to point out that I decided to keep these properties just in case, but they aren't used anywhere. So we can clean them up. I'm happy to hear your opinions

File: packages/experience-builder-sdk/src/core/preview/PrebindingManager.ts [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+    return Object.keys(PrebindingManager.getAllParameterDefinitionsForNodeId(nodeId));
+  }
+
+  /**
+   * Get the highest level overriding parameterId for a given parameterId + nodeId combination
+   * @param parameterId - The parameter to resolve
+   * @param indexedNodeId - The indexed node ID to resolve the parameterId for
+   * @returns string - The resolved parameterId - will be from "root" in patterns or the parent pattern node in experiences
+   */
+  public static getHoistedIdForParameterId(parameterId: string, indexedNodeId: string): string {
```
Comment (Ollie-H) [inline packages/experience-builder-sdk/src/core/preview/PrebindingManager.ts:256]: Do we not need the latest version with the fix applied from this morning?

Comment (Spring3) [reply packages/experience-builder-sdk/src/core/preview/PrebindingManager.ts:256]: Since we don't have the editor functionality at this point, we don't care of the component node ids in the rendering path

Comment (Ollie-H) [reply packages/experience-builder-sdk/src/core/preview/PrebindingManager.ts:256]: Makes sense. Hopefully we don't need to touch it much but the fact they are mostly identical could get confusing why/when they should diverge.

File: packages/experience-builder-sdk/src/utils/prebindingUtils.ts
Comment (SofiaMargariti) [inline packages/experience-builder-sdk/src/utils/prebindingUtils.ts:139]: It's not in your changes, but this will fail if there is no mapping for a specific CT. It's the scenario where I am mapping:
* `CT-A` to:
    * propA
    * propB
* `CT-B` to:
    * prop C
 And my selected source is of type `CT-A`. When resolving mapping for propC, `variableMapping.pathsByContentType[contentTypeId]` will be undefined and this will throw. Adding some optional chaining should cover this case.

**Review by SofiaMargariti — APPROVED** [review]
Sorry it took me a while to test this. The code looks good overall, just an edge-case that I noticed and I've comment above!
